main.tex

```diff
@@ -45,7 +45,26 @@ \section{Complex Predicates~vs.~Complex Objects}
 
 % Applying this translation to all rules that make use of higher order predicates yields a program in first order.
 
-The ASP system \al allows for arbitrary Java objects in place for terms, so a custom class \texttt{ABObjects implements Set} as well as a list of objects (\texttt{List<ABObject>}) may map to $O$.
+\subsection{Translation}
+
+The ASP system \al allows to insert arbitrary Java objects in place of terms.
```

Comment:
OK, so I dont't know how rigorous we want to be here, but my view is that it actually allows terms to *be interpreted as* Java objects. Which is a slightly more formal way of saying it.

In other systems you can have symbolic constants, which are essentially interpreted as themselves (Herbrand style), natural numbers, which are unsurprisingly interpreted as natural numbers, and quoted literal strings, that are then interpreted as strings. With Alpha we extend this and allow any Java object in our domain.